src/main/java/org/jabref/logic/importer/EntryBasedParserFetcher.java

```diff
@@ -54,6 +55,16 @@ default void doPostCleanup(BibEntry entry) {
     default List<BibEntry> performSearch(BibEntry entry) throws FetcherException {
         Objects.requireNonNull(entry);
 
+        try {
+            if (getURLForEntry(entry) == null) {
+                return Collections.emptyList();
+            }
+        } catch (MalformedURLException e) {
```

Comment:
You can combine these catch blocks by using `catch(MalformedURLException|URISyntaxException e)`